public/app/features/dashboard-scene/scene/NavToolbarActions.tsx

```diff
@@ -599,6 +601,25 @@ function useEditingLibraryPanel(panelEditor?: PanelEditor) {
   return isEditingLibraryPanel;
 }
 
+function useVizManagerDirty(panelEditor?: PanelEditor) {
```

Comment:
Because panelEditor is not always defined. Then we run into the problem of conditionally using a hook which could potentially cause issues. The custom hook is always used, and handles the case where panelEditor is undefined internally.

Reply:
I think it's worth capturing in the code comment when this hook is used. Otherwise I would be confused seeing this

Reply:
```suggestion
// This hook handles when panelEditor is not defined to avoid conditionally hook usage
function useVizManagerDirty(panelEditor?: PanelEditor) {
```